src/main/kotlin/apply/application/JudgmentService.kt

```diff
@@ -26,69 +24,54 @@ class JudgmentService(
-        return LastJudgmentResponse(assignment.pullRequestUrl, judgment.lastRecord)
+    fun findLastExampleJudgment(userId: Long, missionId: Long): LastJudgmentResponse? {
+        val assignment = assignmentRepository.findByUserIdAndMissionId(userId, missionId) ?: return null
+        val judgment = judgmentRepository.findByAssignmentIdAndType(assignment.id, JudgmentType.EXAMPLE)
+        return judgment?.let { LastJudgmentResponse(assignment.pullRequestUrl, it.lastRecord) }
     }
 
     fun success(judgmentId: Long, request: SuccessJudgmentRequest) {
         val judgment = judgmentRepository.getById(judgmentId)
         judgment.success(Commit(request.commit), request.passCount, request.totalCount)
-        // TODO: reflect result to evaluation answer
+        judgmentRepository.save(judgment)
     }
 
     fun fail(judgmentId: Long, request: FailJudgmentRequest) {
         val judgment = judgmentRepository.getById(judgmentId)
         judgment.fail(Commit(request.commit), request.message)
+        judgmentRepository.save(judgment)
     }
 
     fun cancel(judgmentId: Long, request: CancelJudgmentRequest) {
         val judgment = judgmentRepository.getById(judgmentId)
         judgment.cancel(Commit(request.commit), request.message)
+        judgmentRepository.save(judgment)
+    }
+
+    fun judgeReal(assignmentId: Long): LastJudgmentResponse {
+        val assignment = assignmentRepository.getById(assignmentId)
+        val mission = missionRepository.getById(assignment.missionId)
+        check(judgmentItemRepository.existsByMissionId(mission.id)) { "자동 채점을 실행할 수 없습니다." }
+        return judge(mission, assignment, JudgmentType.REAL)
     }
 
-    fun findByEvaluationTargetId(evaluationTargetId: Long, type: JudgmentType): JudgmentData? {
-        val evaluationTarget = evaluationTargetRepository.getById(evaluationTargetId)
-        val mission = missionRepository.findByEvaluationId(evaluationTarget.evaluationId) ?: return null
-        val judgmentItem = judgmentItemRepository.findByMissionId(mission.id) ?: return null
-        val assignment = assignmentRepository.findByUserIdAndMissionId(evaluationTarget.userId, mission.id)
-        return assignment
-            ?.let { judgmentRepository.findByAssignmentIdAndType(it.id, type) }
-            .let {
-                JudgmentData(
-                    id = it?.id,
-                    evaluationItemId = judgmentItem.evaluationItemId,
-                    assignmentId = assignment?.id,
-                    judgmentRecord = it?.lastRecord
-                )
-            }
+    fun judge(mission: Mission, assignment: Assignment, judgmentType: JudgmentType): LastJudgmentResponse {
+        var judgment = judgmentRepository.findByAssignmentIdAndType(assignment.id, judgmentType)
+            ?: judgmentRepository.save(Judgment(assignment.id, judgmentType))
+        val commit = assignmentArchive.getLastCommit(assignment.pullRequestUrl, mission.period.endDateTime)
```

Comment:
```suggestion
        val commit = assignmentArchive.getLastCommit(assignment.pullRequestUrl, mission.period.endDateTime)
        var judgment = judgmentRepository.findByAssignmentIdAndType(assignment.id, judgmentType)
            ?: judgmentRepository.save(Judgment(assignment.id, judgmentType))
```

a: 케이스는 적겠지만 불필요한 엔티티 생성을 최소화하기 위해 순서를 변경 할 수 있겠네요 ㅎㅎ